test/model/unreads_test.dart

```diff
@@ -240,6 +240,49 @@ void main() {
     });
   });
 
+  group('isUnread', () {
+    final unreadDmMessage = eg.dmMessage(
+      from: eg.otherUser, to: [eg.selfUser], flags: []);
+    final readDmMessage = eg.dmMessage(
+      from: eg.otherUser, to: [eg.selfUser], flags: [MessageFlag.read]);
+    final unreadChannelMessage = eg.streamMessage(flags: []);
+    final readChannelMessage = eg.streamMessage(flags: [MessageFlag.read]);
+
+    final allMessages = [
+      unreadDmMessage, unreadChannelMessage,
+      readDmMessage,   readChannelMessage,
+    ];
+
+    void doTestCommon(String description, int messageId, bool expected) {
+      test(description, () {
+        prepare();
+        model.oldUnreadsMissing = false;
+        fillWithMessages(allMessages);
+        check(model.isUnread(messageId)).equals(expected);
+      });
+    }
+
+    void doTestOldUnreadsMissing(String description, int messageId, bool? expected) {
+      assert(expected == true || expected == null);
+      test('oldUnreadsMissing; $description', () {
+        prepare();
+        model.oldUnreadsMissing = true;
+        fillWithMessages(allMessages);
+        check(model.isUnread(messageId)).equals(expected);
+      });
+    }
+
+    doTestCommon('unread DM message',      unreadDmMessage.id,      true);
```

Comment:
nit: Should we make the third parameter use a keyword? Just by looking at it it is not so obvious what this is supposed to mean.

Reply:
Sure; made this a named param.